lib/queryBuilder/operations/UpsertGraphOperation.js

```diff
@@ -176,6 +178,23 @@ class UpsertGraphOperation extends InsertGraphOperation {
       { concurrency: this.graph.rootModelClass.concurrency }
     );
   }
+
+  upsertRecursively(builder) {
+    const upsertRecursivelyNodes = this.graph.nodes.filter(it =>
+      it.hasType(UpsertNode.Type.UpsertRecursively)
+    );
+
+    return Promise.map(
+      upsertRecursivelyNodes,
+      node => {
+        return node.upsertModel.constructor
+          .query()
+          .upsertGraph(node.upsertModel, transformOptionsFromPath(node.opt, node.relPathFromRoot))
+          .childQueryOf(builder, true);
+      },
+      { concurrency: this.graph.rootModelClass.concurrency }
```

Comment:
Considering that knex is intending to eventually drop Bluebird usage, is it safe relying on it being there? @koskimas

Reply:
👍 It's also used in similar fashion for `delete`, `relate` and `update` operation currently.